python/cuml/feature_extraction/text.py

```diff
@@ -0,0 +1,578 @@
+        counts.
+    dtype : type, optional
+        Type of the matrix returned by fit_transform() or transform().
+    remove_punctuation : boolean, True by default
+        Remove all characters from string.punctuation before tokenizing.
+    delimiter : str, whitespace by default
+        String used as a replacement for punctuation if remove_punctuation is
+        True and for stop words if stop_words is not None. Typically the
+        delimiting character between words is a good choice. Default is space.
+    Attributes
+    ----------
+    vocabulary_ : nvstrings
+        Array mapping from feature integer indices to feature name.
+    stop_words_ : nvstrings
+        Terms that were ignored because they either:
+          - occurred in too many documents (`max_df`)
+          - occurred in too few documents (`min_df`)
+          - were cut off by feature selection (`max_features`).
+        This is only available if no vocabulary was given.
+    """
+    def __init__(self, input=None, encoding=None, decode_error=None,
+                 strip_accents=None, lowercase=True, preprocessor=None,
+                 tokenizer=None, stop_words=None, token_pattern=None,
+                 ngram_range=(1, 1), analyzer='word', max_df=1.0, min_df=1,
+                 max_features=None, vocabulary=None, binary=False,
+                 dtype=cp.float32, remove_punctuation=True, delimiter=' '):
+        self.preprocessor = preprocessor
+        self.analyzer = analyzer
+        self.lowercase = lowercase
+        self.remove_punctuation = remove_punctuation
+        self.stop_words = stop_words
+        self.max_df = max_df
+        self.min_df = min_df
+        if max_df < 0 or min_df < 0:
+            raise ValueError("negative value for max_df or min_df")
+        self.max_features = max_features
+        if max_features is not None:
+            if not isinstance(max_features, int) or max_features <= 0:
+                raise ValueError(
+                    "max_features=%r, neither a positive integer nor None"
+                    % max_features)
+        self.ngram_range = ngram_range
+        self.vocabulary = vocabulary
+        self.binary = binary
+        self.dtype = dtype
+        self.delimiter = delimiter
+        if dtype not in CUPY_SPARSE_DTYPES:
+            msg = f"Expected dtype in {CUPY_SPARSE_DTYPES}, got {dtype}"
+            raise ValueError(msg)
+
+        sklearn_params = {"input": input,
+                          "encoding": encoding,
+                          "decode_error": decode_error,
+                          "strip_accents": strip_accents,
+                          "tokenizer": tokenizer,
+                          "token_pattern": token_pattern}
+        self._check_sklearn_params(analyzer, sklearn_params)
+
+    def _count_vocab(self, tokenized_df):
+        """Count occurrences of tokens in each document."""
+        # Transform string tokens into token indexes from 0 to len(vocab)
```

Comment:
Small question, do we assume that these the indexes is based on lexicographical ordering, if yes might be worthwhile to add a test and mention in a comment here.

Reply:
Indeed, we assume that the indexes are based on lexicographical ordering. A few tests are testing the vocabulary, and therefore the correct ordering, for instance `test_word_analyzer`.
```python
def test_word_analyzer(ngram_range):
    vec = CountVectorizer(ngram_range=ngram_range).fit(DOCS_GPU)
    ref = SkCountVect(ngram_range=ngram_range).fit(DOCS)
    assert ref.get_feature_names() == vec.get_feature_names().tolist()
```
I'll mention it in a common, thanks!

Reply:
Ahh gotcha, thanks for the pointer , glad that the test all ready exists. 

 Currently in cudf we guarantee these are sorted but that might change so its always good to have a check  like this if that becomes the case.  Thanks 👍